runtime/src/deterministic.rs

```diff
@@ -853,18 +877,31 @@ impl crate::Spawner for Context {
         // Get metrics
         let (label, metric) = spawn_metrics!(self, future);
 
+        // Track parent-child relationship when supervision is requested
+        let parent_children = if self.spawn_config.is_supervised() {
+            Some(self.children.clone())
+        } else {
+            None
+        };
+
         // Set up the task
         let executor = self.executor();
+        self.spawn_config = SpawnConfig::default();
```

Comment:
I landed on the semantics being:
* clone/with_label == retain SpawnConfig
* spawn/spawn_blocking == reset SpawnConfig
* spawn_ref/spawn_blocking_ref == retain SpawnBlocking (felt confusing to modify in this case as we aren't passing the context to an unknowing task)